blivet/devices/storage.py

```diff
@@ -710,11 +710,13 @@ def _set_format(self, fmt):
             elif fmt.min_size and fmt.min_size > self.size:
                 raise errors.DeviceError("device is too small for new format")
 
-        callbacks.format_removed(device=self, fmt=self._format)
-        self._format = fmt
-        self._format.device = self.path
-        self._update_netdev_mount_option()
-        callbacks.format_added(device=self, fmt=self._format)
+        change = self._format != fmt
+        if change:
```

Comment:
Any reason for the extra variable here? Looks to me like ``if self._format != format`` would do and be clear enough on its own.